plugins/default/lib/openshift.py

```diff
@@ -5,18 +5,31 @@
 def oc(*args):
     return check_output(("oc",) + args)
 
+
 def _get_running_pod_names(oc, project):
     # Manually filter Running pods because of a bug in `oc get`, see:
     # https://github.com/kubernetes/kubernetes/issues/29115
-    #return oc("-n", project, "get", "pods", "--show-all=false", "-o", "name")
-    pods = json.loads(oc("-n", project, "get", "pods", "--show-all=false", "-o", "json"))
-    return [p["metadata"]["name"] for p in pods if p["status"]["phase"] == "Running"]
+    # return oc("-n", project, "get", "pods", "--show-all=false", "-o", "name")
+    pods = json.loads(oc("-n", project, "get", "pods","--show-all=false", "-o", "json"))
+    return [p["metadata"]["name"] for p in pods["items"] if p["status"]["phase"] == "Running"]
+
 
 def get_running_pod_names(project):
-    return _get_running_pods(oc, project)
+    return _get_running_pod_names(oc, project)
+
 
 def _exec_in_pods(oc, project, pods, cmd):
-    return [oc("-n", project, "exec", name, "--", *cmd) for name in pods]
+    return [[name, oc("-n", project, "exec", name, "--", *cmd)] for name in pods]
```

Comment:
This is strange. We pass in the pod names as argument, normally we don't need the names in the output.
The problem here is where this function is called. We know the pod names there.